internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -307,6 +307,50 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				},
 			},
 
+			"azure_monitor_profile": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"metrics": {
+							Type:     pluginsdk.TypeList,
+							MaxItems: 1,
+							Optional: true,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"enabled": {
+										Type:     pluginsdk.TypeBool,
+										Required: true,
```

Comment:
we don't expose `enabled` as a boolean, we can use the presence of the `azure_monitor` block to indicate this

since there's only one block within this, we can also remove the `azure_monitor_profile` block instead and expose this as `azure_monitor_kubernetes_metrics` (or something) instead to remove all the nesting?